src/KOKKOS/pair_lj_spica_coul_long_kokkos.cpp

```diff
@@ -0,0 +1,574 @@
+
+  } else if (ljt == LJ9_6) {
+
+    const F_FLOAT r3inv = r2inv*sqrt(r2inv);
+    const F_FLOAT r6inv = r3inv*r3inv;
+    return r6inv*(lj_1*r3inv - lj_2) * r2inv;
+
+  } else if (ljt == LJ12_6) {
+
+    const double r6inv = r2inv*r2inv*r2inv;
+    return r6inv*(lj_1*r6inv - lj_2) * r2inv;
+
+  } else if (ljt == LJ12_5) {
+
+    const F_FLOAT r5inv = r2inv*r2inv*sqrt(r2inv);
+    const F_FLOAT r7inv = r5inv*r2inv;
+    return r5inv*(lj_1*r7inv - lj_2) * r2inv;
+
+  }
+  if (ljt!=LJ12_4 && ljt!=LJ9_6 && ljt!=LJ12_6 && ljt!=LJ12_5) return 0.0;*/
+  const F_FLOAT r4inv=r2inv*r2inv;
+  const F_FLOAT r6inv=r2inv*r4inv;
+  const F_FLOAT a = ljt==LJ12_4?r4inv:(ljt==LJ12_5?r4inv*sqrt(r2inv):r6inv);
+  const F_FLOAT b = ljt==LJ12_4?r4inv:(ljt==LJ9_6?1.0/sqrt(r2inv):(ljt==LJ12_5?r2inv*sqrt(r2inv):r2inv));
+  return a* ( lj_1*r6inv*b - lj_2 * r2inv);
+}
+
+/* ----------------------------------------------------------------------
+   compute pair potential energy between atoms i and j
+   ---------------------------------------------------------------------- */
+
+template<class DeviceType>
+template<bool STACKPARAMS, class Specialisation>
+KOKKOS_INLINE_FUNCTION
+F_FLOAT PairLJSPICACoulLongKokkos<DeviceType>::
+compute_evdwl(const F_FLOAT& rsq, const int& i, const int&j, const int& itype, const int& jtype) const {
+  (void) i;
+  (void) j;
+  const F_FLOAT r2inv = 1.0/rsq;
+  const int ljt = (STACKPARAMS?m_params[itype][jtype].lj_type:params(itype,jtype).lj_type);
+
+  const F_FLOAT lj_3 =  (STACKPARAMS?m_params[itype][jtype].lj3:params(itype,jtype).lj3);
+  const F_FLOAT lj_4 =  (STACKPARAMS?m_params[itype][jtype].lj4:params(itype,jtype).lj4);
+  const F_FLOAT offset =  (STACKPARAMS?m_params[itype][jtype].offset:params(itype,jtype).offset);
+
+  if (ljt == LJ12_4) {
+    const F_FLOAT r4inv=r2inv*r2inv;
+
+    return r4inv*(lj_3*r4inv*r4inv - lj_4) - offset;
+
+  } else if (ljt == LJ9_6) {
+    const F_FLOAT r3inv = r2inv*sqrt(r2inv);
+    const F_FLOAT r6inv = r3inv*r3inv;
+    return r6inv*(lj_3*r3inv - lj_4) - offset;
+
+  } else if (ljt == LJ12_6) {
+    const double r6inv = r2inv*r2inv*r2inv;
+    return r6inv*(lj_3*r6inv - lj_4) - offset;
+
+  } else if (ljt == LJ12_5) {
+    const F_FLOAT r5inv = r2inv*r2inv*sqrt(r2inv);
+    const F_FLOAT r7inv = r5inv*r2inv;
+    return r5inv*(lj_3*r7inv - lj_4) - offset;
+  } else
+    return 0.0;
+}
+
+/* ----------------------------------------------------------------------
+   compute coulomb pair force between atoms i and j
+------------------------------------------------------------------------- */
+
+template<class DeviceType>
+template<bool STACKPARAMS,  class Specialisation>
+KOKKOS_INLINE_FUNCTION
+F_FLOAT PairLJSPICACoulLongKokkos<DeviceType>::
+compute_fcoul(const F_FLOAT& rsq, const int& /*i*/, const int&j,
+              const int& /*itype*/, const int& /*jtype*/,
+              const F_FLOAT& factor_coul, const F_FLOAT& qtmp) const {
+
+/*
+
+        r2inv = 1.0 / rsq;
```

Comment:
Please remove the commented out code, or justify why it is helpful

Reply:
done